unity/Assets/Scripts/ImageSynthesis/Shaders/Depth.shader

```diff
@@ -51,24 +56,21 @@ Shader "Hidden/Depth" {
              fixed4 frag(output o) : COLOR
              {
                  // depth01 = pow(LinearEyeDepth(depth01), _DepthLevel);
-                 float depth01 = Linear01Depth(UNITY_SAMPLE_DEPTH(tex2D(_CameraDepthTexture, o.uv)));
-
-                 // float lowBits = depth01 / 2;
-                 // float medBits = depth01 / 4;
-                 // float highBits = depth01 / 8;
+                 float multiplier = _FarClipPlane - _NearClipPlane;
+                 float depth01 = Linear01Depth(UNITY_SAMPLE_DEPTH(tex2D(_CameraDepthTexture, o.uv))) * multiplier;
```

Comment:
If we want to stick with our 3D "units" ("meters"), as opposed to normalized, we may as well just use the`LinearEyeDepth` function instead, which returns the depth like that. Where 1 is a surface that is 1 unit from the camera’s pivot along the camera’s z axis.

Reply:
I just tried `LinearEyeDepth`: 
```
float depth01 = LinearEyeDepth(UNITY_SAMPLE_DEPTH(tex2D(_CameraDepthTexture, o.uv)));
```
and I was getting a result that didn't include the near plane offset - If say the agent was 3.0m from a wall, the depth value was getting reported back as 3.0m instead of 2.985 (what we currently get back).

As well, I prefer the way we have it currently because it gives us the flexibility to push the conversion into the depth shader as a parameter in the future in the form of the `multiplier`.

Reply:
Makes sense